24.07.00 grapes js web builder integration

AlexanderBlanchardAC opened the pull request:
Test Plan:

1.Enable the Web Builder Module by navigating to System Administration -> Modules -> WebBuilder.
2.Once enabled, notice that there are now two new options in Web Builder: Grapes Pages and Templates.
3.Click on templates and add new. Give your new template a name and save.
4.Click 'Open in Editor' to open your template in the Grapes Editor.
5.Add some content using the grapes editor.
6.Click the save button and then navigate back to your Web Builder Section.
7.Click on Grapes Pages and add a new page.
8.Fill in your page details and select a template. Click save.
9.Use the 'Open in Editor' button to open the page in the editor. Note it contains the html information form your template.
10.Add to your page and click save.
11.Reload the page and note that your new page data is persistent.
12.Navigate to your page using the URL you gave in the Grapes Page set up.

AlexanderBlanchardAC committed:
feat: update db and system API

Create tables in DB for Grapes Pages and add grapes updates to system API

AlexanderBlanchardAC committed:
feat: Add Section in Web Builder Module

Add a section for Grapes Pages and Another for grapes Temapltes in Web Builder Module.

AlexanderBlanchardAC committed:
feat: Add DB Objects

Add objects for the Grapes Page and Templates

AlexanderBlanchardAC committed:
feat: add grapespages to index.php

AlexanderBlanchardAC committed:
feat: Add .tpl files and CDNs

Add .tpl files for grapes pages and templates and add CDNs fro grapes.js

AlexanderBlanchardAC committed:
feat: update admin.js to nclude grapes pages

AlexanderBlanchardAC committed:
feat: Include grapes pages in layout.tpl

AlexanderBlanchardAC committed:
feat: Handle indexing of Grapes Pages

Update Java files and book cover files to handle the indexing of
Grapes Pages.

AlexanderBlanchardAC committed:
chore: Rebuild jar file

Rebuild jar file after altering web indexing java files.

AlexanderBlanchardAC committed:
feat: Add Grapes Pages to website searcher

AlexanderBlanchardAC committed:
feat: Update WebBuilder AJAX

Update WebBuilder AJAZX to allow the saving and loading of Grapes Pages and
Grapes Temapltes.

AlexanderBlanchardAC committed:
fix: Prevent delete function error

Alter delete function in GrapesPage.php to ensure it is compatible with
the parent method.

AlexanderBlanchardAC committed:
chore: Update changelog

K-Alette committed:
don't use accessible browse category style for search results

catsoup11789 committed:
Merge pull request #1872 from K-Alette/24.06.00

don't use accessible browse category style for search results

AlexanderBlanchardAC committed:
Merge branch '24.07.00' of https://github.com/mdnoble73/aspen-discovery into 24.07.00_grapes_js_web_builder_integration

AlexanderBlanchardAC committed:
feat: Allow Grapes Pages to Index

Add Grapes Pages back into the fucntion to check for Web Builder url Aliases so that
Grapes Pages index.

AlexanderBlanchardAC committed:
refactor: Adjust breadcrumbs

Adjust breadcrumbs for ease of navigaiton.

AlexanderBlanchardAC committed:
feat: Allow navigation to in page view

Add action to each grapes page object in list to allow a view of the created page outside of the editor - navigating to its urlAlias.

Jacobomara901 commented on code/web/services/WebBuilder/GrapesPages.php:
```diff
@@ -0,0 +1,115 @@
+		return 'web_builder';
+	}
+
+    public function canAddNew(){
```
indenting here doesn't match the above

Jacobomara901 commented on code/web/services/WebBuilder/Templates.php:
```diff
@@ -0,0 +1,129 @@
+		return 'Template';
+	}
+
+	function getToolName(): string {
```
indenting hereeee

Jacobomara901 replied:
same thing below :)

Jacobomara901 commented on code/web/sys/WebBuilder/GrapesPage.php:
```diff
@@ -0,0 +1,315 @@
+	public $title;
+	public $urlAlias;
+	public $teaser;
+    public $pageType;
```
same here, indenting

Jacobomara901 commented on code/web/sys/WebBuilder/GrapesPage.php:
```diff
@@ -0,0 +1,315 @@
+			}
+			unset($this->_libraries);
+		}
+    }
```
indenting

Jacobomara901 replied:
same belowwww

Jacobomara901 commented on code/web/sys/WebBuilder/GrapesPage.php:
```diff
@@ -0,0 +1,315 @@
+        return false;
+    }
+
+	function getAdditionalObjectActions($existingObject): array{
```
indenting here

Jacobomara901 commented on code/web/interface/themes/responsive/WebBuilder/createTemplatejs.tpl:
```diff
@@ -0,0 +1,153 @@
+<!DOCTYPE html>
+
+<head>
+    <title>Create Template</title>
```
indenting here

Jacobomara901 commented on code/web/interface/themes/responsive/WebBuilder/grapesPage.tpl:
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
```
the indenting for the tags here could use some love

Jacobomara901 commented on code/web/interface/themes/responsive/WebBuilder/grapesjs.tpl:
```diff
@@ -0,0 +1,151 @@
+        command: 'save-as-page',
+        attributes: { title: 'Save as Page' }
+      }]);
+      editor.Commands.add('save-as-page', {
```
this editor command and the one below have different indenting

Jacobomara901 commented:
Hey, this is really cool. Great to see how far it's come from when I was last working on it with you.
Couple of things I noted:
Having to save the template, leave the page then go back in to edit is a little annoying and perhaps could be streamlined.
The save button for grapes is easily missed and is not super obvious at first glance. (obviously I know where it is anyway, but for a new user this could be a little annoying)
I don't think we should HAVE to use one of the saved templates for creating a grapes page. I think this should be an option and if you choose not to use a template you should be dropped in to a blank page, doesn't seem to be an option for that currently. Additionally, this is the only required field for creating the page.
The map section in grapes displays differently in the editor to how it displays on a page.
The text in the grapes editor is still very small and sometimes hard to see.
I see you've added record driver stuff for grapes in the code. How do I get records to display on my grapes pages, that's super cool.

Code looks really good, the only comments I've made are where I've spotted some inconsistencies with styling, specifically the indenting of lines. Other than that I'm not seeing any issues bud. I've put comments on the lines where I noticed the inconsistencies for easy fixing for you, though there are probably some I also missed

AlexanderBlanchardAC committed:
refactor: Fix getAdditionalObjectActions

Fix the indentation of getAdditionalObjectActions function

AlexanderBlanchardAC committed:
refactor: Address indentation issues

AlexanderBlanchardAC commented:
Cheers Jacob, I have addressed the indentation issues. I have added the Grapes general feedback to my Excel Sheet. Some good points. I will add to further improvements branch.

AlexanderBlanchardAC committed:
refactor: Change permissions for grapes page
